### ISSUE TITLE
False negative: `clippy::precedence` not raised in some cases

### ISSUE THREAD

**Issue by saveriomiroddi** [rust-lang/rust-clippy#5925]
When running clippy on the following code:

```rust
    let val1 = -0.0_f64.cos();
    let val2 = -0.0_f64.cos().ceil();
    println!("{:3}, {:3}", val1, val2);
```

I expected to see this happen: the `warning: unary minus has lower precedence than method call` should be reported on the two assignment lines.

Instead, this happened: the warning is raised on ly on the first assignment.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (d3fb005 2020-07-31)
- `rustc -Vv`:
  ```
rustc 1.45.2 (d3fb005a3 2020-07-31)
binary: rustc
commit-hash: d3fb005a39e62501b8b0b356166e515ae24e2e54
commit-date: 2020-07-31
host: x86_64-unknown-linux-gnu
release: 1.45.2
LLVM version: 10.0
  ```

<details><summary>Backtrace</summary>
  <p>
  
  ```
warning: unary minus has lower precedence than method call
  --> playground/src/main.rs:11:16
   |
11 |     let val1 = -0.0_f64.cos();
   |                ^^^^^^^^^^^^^^ help: consider adding parentheses to clarify your intent: `-(0.0_f64.cos())`
   |
   = note: `#[warn(clippy::precedence)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#precedence

warning: 1 warning emitted

  ```
  
  </p>
</details>

**Comment by saveriomiroddi** [comment]
Original Rust programming language forum [discussion](https://users.rust-lang.org/t/why-f64-sin-ceil-give-inconsistent-results-based-on-the-invocation/).

**Comment by ebroto** [comment]
Closing as a duplicate of #5924
